cli: TestStageVersionCheck failed

cli.TestStageVersionCheck [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11335380?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11335380?buildTab=artifacts#/) on master @ [7c5146204558a97d5412f05080807586daf336b2](https://github.com/cockroachdb/cockroach/commits/7c5146204558a97d5412f05080807586daf336b2):


```
=== RUN   TestStageVersionCheck
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/6c689cc523db9d8596d03c0a6ef213e4/logTestStageVersionCheck1531780939
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:118: test server using tenant; see comment at top of test_server_shim.go for details.
*
* WARNING: test tenant requested by configuration, but code organization prevents start!
* OSS binaries do not include enterprise features
*
*
* WARNING: test tenant requested by configuration, but code organization prevents start!
* OSS binaries do not include enterprise features
*
*
* WARNING: test tenant requested by configuration, but code organization prevents start!
* OSS binaries do not include enterprise features
*
*
* WARNING: test tenant requested by configuration, but code organization prevents start!
* OSS binaries do not include enterprise features
*
    debug_recover_loss_of_quorum_test.go:378: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/cli/debug_recover_loss_of_quorum_test.go:378
        	Error:      	Received unexpected error:
        	            	rpc error: code = Unknown desc = initial connection heartbeat failed: grpc: client cluster ID "c7f496dd-57ac-4e1b-9b8d-2224e10e7394" doesn't match server cluster ID "213e90ba-5b80-4e51-a0d1-a247b458aebe" [code 2/Unknown]
        	Test:       	TestStageVersionCheck
        	Messages:   	force local must fix incorrect version
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/6c689cc523db9d8596d03c0a6ef213e4/logTestStageVersionCheck1531780939
--- FAIL: TestStageVersionCheck (4.55s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStageVersionCheck.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-30646

--- a/pkg/cli/debug_recover_loss_of_quorum_test.go
+++ b/pkg/cli/debug_recover_loss_of_quorum_test.go
@@ -328,6 +328,9 @@ func TestStageVersionCheck(t *testing.T) {
 	})
 	defer c.Cleanup()
 
+	listenerReg := listenerutil.NewListenerRegistry()
+	defer listenerReg.Close()
+
 	storeReg := server.NewStickyVFSRegistry()
 	tc := testcluster.NewTestCluster(t, 4, base.TestClusterArgs{
 		ReplicationMode: base.ReplicationManual,
@@ -343,6 +346,7 @@ func TestStageVersionCheck(t *testing.T) {
 				},
 			},
 		},
+		ReusableListenerReg: listenerReg,
 	})
 	tc.Start(t)
 	defer tc.Stopper().Stop(ctx)

